packages/mui-joy/src/Autocomplete/Autocomplete.test.tsx

```diff
@@ -34,6 +34,58 @@ function checkHighlightIs(listbox: HTMLElement, expected: string | null) {
   }
 }
 
+type TestOnScrollToBottom = {
+  reason: 'mouse' | 'keyboard';
+  scrollToBottomCallCount: number;
+  onScrollToBottom: ReturnType<typeof spy>;
+};
+
+function testOnScrollToBottom({
+  reason,
+  scrollToBottomCallCount,
+  onScrollToBottom,
+}: TestOnScrollToBottom) {
+  const textbox = screen.getByRole('combobox');
+  const listbox = screen.getByRole('listbox');
+
+  if (reason === 'mouse') {
+    const scrollTop = listbox.scrollHeight - listbox.offsetHeight;
+
+    if (scrollTop > 0) {
+      fireEvent.scroll(listbox, {
+        target: {
+          scrollTop,
+        },
+      });
+    }
+
+    if (scrollToBottomCallCount > 0) {
+      expect(listbox.scrollTop).to.greaterThan(0);
+    } else {
+      expect(listbox.scrollTop).to.equal(0);
+    }
+    expect(onScrollToBottom.callCount).to.equal(scrollToBottomCallCount);
```

Comment:
Since these same assertions are running on both the cases you can move this outside of the `reason`'s `if..else` block.